regions/io/core.py

```diff
@@ -381,10 +384,10 @@ def to_fits(self):
                 x.append(shape.coord[0].value)
                 y.append(shape.coord[1].value)
                 if shape.region_type in ['circle', 'circleannulus', 'point']:
-                    r.append([x.value for x in shape.coord[2:]])
```

Comment:
Comparing this line to that two above, a few issues:
(1) You're re-using the variable `x`.  Even though it's in a different scope, you should avoid this; it can lead to confusion.
(2) Should this be `float(x.value)`?  It looks like `shape.coord` is still a `Quantity`.

Reply:
Nice catch! But the problem is `r` (radius) contains a scalar length for pixels regions. So, if we convert list of regions to  a `ShapeList` object , it will mostly contain a float (`x.value` wont work).
There is a case it can be `dimensionless` quantity in the `Shape` object which when typecasted by `float()` converts it into a scalar type. (This is the case when I try to parse from the sample file, the `FITSRegionParser` gets us the `ShapeList` where the radius may be a `dimensionless` quantity depending on the unit of the `R` column.)

Reply:
OK, I understand.  Is it definitely safe to do float(x), then?  If so, that's fine; I haven't digested the whole context of this line so I'm not sure.

Reply:
The bottomline is `shape.coord[2:]` would be containing either `float` or `dimensionless` quantites. So, it is safe.